### PR TITLE
Fix test case for finding wallets by name pattern and update repository methods (#251)

### PR DESCRIPTION
**Pull request by sylvvester**
…ry methods

#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2610; New code is covered with tests and coverage requirements are met.
&#x2610; Build is not failing.
&#x2610; All tests are passing.
&#x2610; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2610; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2610; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2610; Correct unit test name - suffix 'Test' in the end.
&#x2610; Correct integration test name - suffix 'IT' in the end.
&#x2610; Positive and negative test paths have been implemented.

### CODE REVIEW

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java [hunk 1]
```diff
@@ -117,22 +117,29 @@ void shouldNotReturnWallet_WhenWalletByIdNotFound() {
     //fixme, new issue, required improve method for wallets
     @Test
     @DisplayName("when finding wallet by name should return all wallets contains this name pattern")
-    @Disabled
     void shouldReturnAllWalletsContainsNamePattern_WhenFindingWalletByName() {
         // given
         String walletNameSearched = "Family";
         List<Wallet> walletList = createListOfWalletsByName("Family wallet", "Common Wallet", "Smith Family Wallet");
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java:123]: At least one wallet name must contains 'family' lower case.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java [hunk 2]
```diff
@@ -117,22 +117,29 @@ void shouldNotReturnWallet_WhenWalletByIdNotFound() {
...
+        given(walletRepository.findAllByNameIsContainingIgnoreCase(walletNameSearched))
+                .willReturn(walletList.stream()
+                                .filter(wallet -> wallet.getName().contains(walletNameSearched))
+                                .toList());
         walletList.forEach(wallet -> given(walletModelMapper.mapWalletEntityToWalletDTO(wallet)).willReturn(
-                walletListDTO.stream().filter(walletDTO -> Objects.equals(wallet.getName(), walletDTO.name())).findAny().orElse(null)));
-
+                walletListDTO
+                        .stream()
+                        .filter(walletDTO -> Objects.equals(wallet.getName(), walletDTO.name()))
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java:134]:
```suggestion
                        .filter(wallet -> wallet.getName().toUpperCase().contains(walletNameSearched.toUpperCase()))
```
In this method, we should compare the names case-insensitive.

**Review by mirekgab — CHANGES_REQUESTED** [review]

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java (hunk 2 above)
Comment (mirekgab) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletGetServiceImplTest.java:134]: Please fix tests in `FindAllWalletsByNameCaseSensitiveIT` class.
In the method `createListTestWallets` add another wallet 
```java
        final Wallet fifthWallet = Wallet.builder()
                .user(user)
                .creationDate(Instant.now())
                .name("Wallet Test")
                .build();
```

**Review by mirekgab — APPROVED** [review]

**Review by Iwona007 — APPROVED** [review]

**Review by dawciobiel — APPROVED** [review]
Looks good to me

**Pull request merged**